### ISSUE TITLE
All Loot Drops not working for some mob drops

### ISSUE THREAD

**Issue by antermin** [ricksouth/serilum-mc-mods#1436]
## **Information**
Minecraft version: 1.18.2
Modloader: Forge
Forge version: 40.1.85
Environment: Singleplayer

Mod name: All Loot Drops
Mod version: 2.6 (1.18.2-2.6)

## **Description**
While the mod does work for some mobs such as Witch and Creeper, some mob drops (apparently, conditional drops which are only available when the mob is killed by a player) are not affected by the mod, e.g.
* Blaze is not dropping at least 1 Blaze Rod
* (Cave) Spider is not dropping at least 1 Spider Eye
* Zombie is not dropping at least 1 Iron Ingot, Carrot and Potato

I tested in Creative mode by using Spawn Eggs and killing the mobs with an unenchanted Netherite Sword.

## **Config file**
```
[General]
	#If for example Iron Golems drop more iron than the default lootQuantity setting, keep that original amount instead of nerfing it.
	keepOriginalLootQuantityIfHigher = true
	#Determines the amount of loot dropped by each mob.
	#Range: 1 ~ 128
	lootQuantity = 1
	#If enabled, the looting enchant has a chance to increase the quantity of loot dropped. Per level a roll is done with the chance from 'lootingEnchantExtraQuantityChancePerLevel'. If the roll succeeds, 1 is added to the loot amount.
	lootingEnchantAffectsQuantity = true
	#The chance a roll succeeds in adding 1 to the total loot amount.
	#Range: 0.0 ~ 1.0
	lootingEnchantExtraQuantityChance = 0.5
```

**Comment by James103** [comment]
Does this work if the mobs are killed using the sword generated by the following command?
`give @s netherite_sword{Enchantments:[{id:"looting",lvl:255}]}`

The high Looting level may elevate rare drop chances to be as common as common drops. If the rare drops still aren't being dropped, then something must be really wrong with the mod.

**Comment by antermin** [comment]
Using the sword generated by your command, I get 275 Blaze Rods from killing 3 Blazes.

**Comment by James103** [comment]
Does that same sword work on zombies and spiders?
(For zombies, does it drop at least an iron ingot, carrot, and potato most of the time?)

**Comment by antermin** [comment]
Yes, with the same sword, zombies drop 1 x (Iron Ingot or Carrot or Potato) along with many rotten flesh, and for spiders it drops many spider eyes.

**Comment by Jetpack-Cat** [comment]
I gave zombies swords with incontrol mod and they're not dropping them, the mod basically doesn't do anything?

**Comment by EightBitBBQ** [comment]
Modversion is at 3.1 now. Do you still have this issue?

**Comment by SophieScott12** [comment]
Is this going to get a fix?  It's still an issue in mod version 3.4 on MC 1.20.1.

**Comment by staltux** [comment]
v3.3 still have this issue, zombies not dropping carrots and iron
